Source/Core/Core.js

```diff
@@ -397,6 +398,7 @@ Object.extend({
 	},
 
 	clone: function(object){
+		if(Object.create) return Object.create(object);
```

Comment:
The `Object.create` function creates a new object that inherits directly from the one passed as the first argument, in opposite the `Object.clone` creates a new object with always the base `Object` prototype, this is actually changing the `Object.clone` behavior. https://jsfiddle.net/AmraniCh/ed5krqb9/

Reply:
@AmraniCh: In your fiddle, you should use `Array.clone` with arrays and not `Object.clone`
https://jsfiddle.net/vu5xt0ed/

Reply:
@SergioCrisostomo Actually it `Object.clone`, I'm just making a test with the array type. 
The Mootools `Object.clone` always clone an object and return a new one with the `Object` prototype, if we add the `return Object.create(object)` then the `Object.clone` will use the object passed as an argument (can be an array, function ..) as a prototype of the newly created object, which is not the native behavior of the `Object.clone`.